File: attentive-android-sdk/src/test/java/com/attentive/androidsdk/FactoryMocks.java
package com.attentive.androidsdk;

import static org.mockito.ArgumentMatchers.any;
import static org.mockito.Mockito.spy;

import com.fasterxml.jackson.databind.ObjectMapper;
import okhttp3.OkHttpClient;
import org.mockito.MockedStatic;
import org.mockito.Mockito;

public class FactoryMocks implements AutoCloseable {
    private final MockedStatic<ClassFactory> classFactoryMockedStatic;
    private final PersistentStorage persistentStorage;
    private final VisitorService visitorService;
    private final OkHttpClient okHttpClient;
    private final AttentiveApi attentiveApi;
    private final ObjectMapper objectMapper;

    private FactoryMocks(MockedStatic<ClassFactory> classFactoryMockedStatic,
                         PersistentStorage persistentStorage,
                         VisitorService visitorService, OkHttpClient okHttpClient,
                         AttentiveApi attentiveApi, ObjectMapper objectMapper) {
        this.classFactoryMockedStatic = classFactoryMockedStatic;
        this.persistentStorage = persistentStorage;
        this.visitorService = visitorService;
        this.okHttpClient = okHttpClient;
        this.attentiveApi = attentiveApi;
        this.objectMapper = objectMapper;
    }

    public static FactoryMocks mockFactoryObjects() {
        MockedStatic<ClassFactory> classFactoryMockedStatic = Mockito.mockStatic(ClassFactory.class);

        PersistentStorage persistentStorage = Mockito.mock(PersistentStorage.class);
        classFactoryMockedStatic.when(() -> ClassFactory.buildPersistentStorage(any())).thenReturn(persistentStorage);

        VisitorService visitorService = Mockito.mock(VisitorService.class);
        classFactoryMockedStatic.when(() -> ClassFactory.buildVisitorService(any())).thenReturn(visitorService);

        ObjectMapper objectMapper = spy(new ObjectMapper());
        classFactoryMockedStatic.when(ClassFactory::buildObjectMapper).thenReturn(objectMapper);

        OkHttpClient okHttpClient = Mockito.mock(OkHttpClient.class);
        classFactoryMockedStatic.when(ClassFactory::buildOkHttpClient).thenReturn(okHttpClient);

        AttentiveApi attentiveApi = Mockito.mock(AttentiveApi.class);
        classFactoryMockedStatic.when(() -> ClassFactory.buildAttentiveApi(any(), any())).thenReturn(attentiveApi);

        return new FactoryMocks(classFactoryMockedStatic, persistentStorage, visitorService, okHttpClient, attentiveApi,
            objectMapper);
    }

    public PersistentStorage getPersistentStorage() {
        return this.persistentStorage;
    }

    public VisitorService getVisitorService() {
        return this.visitorService;
    }

    public OkHttpClient getOkHttpClient() {
        return okHttpClient;
    }

    public AttentiveApi getAttentiveApi() {
        return attentiveApi;
    }

    public ObjectMapper getObjectMapper() {
        return objectMapper;
    }

    @Override
    public void close() {
        this.classFactoryMockedStatic.close();
    }
}

File: attentive-android-sdk/src/test/java/com/attentive/androidsdk/FactoryMocks.kt
package com.attentive.androidsdk

import com.attentive.androidsdk.AttentiveApi
import com.attentive.androidsdk.ClassFactory.buildAttentiveApi
import com.attentive.androidsdk.ClassFactory.buildOkHttpClient
import com.attentive.androidsdk.ClassFactory.buildPersistentStorage
import com.attentive.androidsdk.ClassFactory.buildVisitorService
import com.attentive.androidsdk.PersistentStorage
import com.attentive.androidsdk.VisitorService
import okhttp3.OkHttpClient
import org.mockito.ArgumentMatchers
import org.mockito.MockedStatic
import org.mockito.MockedStatic.Verification
import org.mockito.Mockito
import org.mockito.kotlin.any

class FactoryMocks private constructor(
    private val classFactoryMockedStatic: MockedStatic<ClassFactory>,
    val persistentStorage: PersistentStorage,
    val visitorService: VisitorService, val okHttpClient: OkHttpClient,
    val attentiveApi: AttentiveApi
) : AutoCloseable {
    override fun close() {
        classFactoryMockedStatic.close()
    }

    companion object {
        fun mockFactoryObjects(): FactoryMocks {
            val classFactoryMockedStatic = Mockito.mockStatic(
                ClassFactory::class.java
            )

            val persistentStorage = Mockito.mock(
                PersistentStorage::class.java
            )
            classFactoryMockedStatic.`when`<Any> {
                buildPersistentStorage(
                    any()
                )
            }.thenReturn(persistentStorage)

            val visitorService = Mockito.mock(VisitorService::class.java)
            classFactoryMockedStatic.`when`<Any> {
                buildVisitorService(
                    any()
                )
            }.thenReturn(visitorService)

            val okHttpClient = Mockito.mock(OkHttpClient::class.java)
            classFactoryMockedStatic.`when`<Any> {
                buildOkHttpClient(
                    any(),
                    any()
                )
            }.thenReturn(okHttpClient)

            val attentiveApi = Mockito.mock(AttentiveApi::class.java)
            classFactoryMockedStatic.`when`<Any> {
                buildAttentiveApi(
                    any()
                )
            }.thenReturn(attentiveApi)

            return FactoryMocks(
                classFactoryMockedStatic,
                persistentStorage,
                visitorService,
                okHttpClient,
                attentiveApi
            )
        }
    }
}
